src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -336,4 +325,24 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
             return file;
         };
     }
+
+    /**
+     * Extract the bucket from a gs path.
+     */
+    public static String extractBucketFromPath(final String path) {
```

Comment:
This code (and `extractFilePathInBucket`) duplicates the code in `getBlobFromGsPath`. Making a common method that returns the bucket and path for a gs URL (as well as validation) would be better.